test/javascript_prelude/main.mjs

```diff
@@ -256,6 +256,15 @@ let weak_set = new WeakSet([map, set]);
 assertEqual(weak_set, weak_set);
 assertNotEqual(new WeakSet([map, set]), new WeakSet([map, set]));
 
+// RegExp are not equal unless they have reference equality
```

Comment:
They should be compared structurally, not by reference.

Reply:
That's how I initially implemented it (comparing the  `source` and `flags` of each `RegExp` ), but second guessed by self and thought they should be compared referentially - reverted back to that approach 👍